sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/CatalystSqlParser.scala

```diff
@@ -16,11 +16,13 @@
  */
 package org.apache.spark.sql.catalyst.parser
 
+import org.apache.spark.sql.internal.SQLConf
+
 /**
  * Concrete SQL parser for Catalyst-only SQL statements.
  */
 class CatalystSqlParser extends AbstractSqlParser {
```

Comment:
I checked all the caller side, it's not test only, but is only used from simple tasks like parsing table schema or table identifier, which doesn't need SQLConf at all.

Reply:
It needs SQLConf for ANSI mode. It does make a difference for what we allow in terms of keywords. It is not super problematic because you can typically fix this by quoting identifiers, the only problem is that within the same session you could have different parsing behavior.

Reply:
It's also used for interval literal parsing